frontend/src/pages/AllRunsList.tsx

```diff
@@ -103,6 +116,40 @@ class AllRunsList extends Page<{}, AllRunsListState> {
       this.props.history.push(RoutePage.NEW_RUN + searchString);
     }
   }
+
+  private async _archiveDialogClosed(confirmed: boolean): Promise<void> {
+    if (confirmed) {
+      const unsuccessfulIds: string[] = [];
+      const errorMessages: string[] = [];
+      await Promise.all(this.state.selectedIds.map(async (id) => {
+        try {
+          await Apis.runServiceApi.archiveRun(id);
+        } catch (err) {
+          unsuccessfulIds.push(id);
+          const errorMessage = await errorToMessage(err);
+          errorMessages.push(`Deleting run failed with error: "${errorMessage}"`);
+        }
+      }));
+
+      const successfulObjects = this.state.selectedIds.length - unsuccessfulIds.length;
+      if (successfulObjects > 0) {
+        this.props.updateSnackbar({
+          message: `Successfully archived ${successfulObjects} run${s(successfulObjects)}!`,
+          open: true,
+        });
+        this.refresh();
```

Comment:
any potential issues with not awaiting this if some archives succeeded and others didn't? I think it should be alright

Reply:
The only thing I can think of is a potential race condition between the `refresh` method calling `showErrorDialog` if it fails, and the below call to `showErrorDialog` if an archive call fails. It's not obvious to me, however, if awaiting fixes that, so I'll hold off on making any changes there for now.